scripts/scraper-ng/rules/html-warn-macros.js

```diff
@@ -1,28 +1,48 @@
-const rule = require("unified-lint-rule");
 const visit = require("unist-util-visit-parents");
 
 /**
  * Issue a warning for each macro call (except for the array of macro names in
  * `allowedMacros`)
  */
-function warnOnMacros(tree, file, allowedMacros) {
+function attacher(allowedMacros) {
   allowedMacros = Array.isArray(allowedMacros) ? allowedMacros : [];
 
-  visit(
-    tree,
-    node =>
-      node.type === "text" &&
-      node.data &&
-      node.data.macroName &&
-      !allowedMacros.includes(node.data.macroName),
-    (node, ancestors) => {
-      // Because macro nodes are generated, they don't have position
-      // information. The parent node's position is used instead.
-      const parent = ancestors[ancestors.length - 1];
-      const message = file.message(`Macro: ${node.data.macroName}`, parent);
-      message.note = `With arguments: ${JSON.stringify(node.data.macroParams)}`;
-    }
-  );
+  return function warnOnMacros(tree, file) {
+    visit(
+      tree,
+      node =>
+        node.type === "text" &&
+        node.data &&
+        node.data.macroName &&
+        !allowedMacros.includes(node.data.macroName),
```

Comment:
Thanks to a comment @peterbe made, I realize that this needs to use the normalized form of the macro name, otherwise we generate a warning for `compat` when `allowedMacros` lists `Compat`.

Reply:
Fixed!